clients/src/main/java/org/apache/kafka/common/metrics/stats/Avg.java

```diff
@@ -36,13 +36,13 @@ protected void update(Sample sample, MetricConfig config, double value, long now
 
     @Override
     public double combine(List<Sample> samples, MetricConfig config, long now) {
-        double total = 0.0;
+        double total = 0;
         long count = 0;
         for (Sample s : samples) {
             total += s.value;
             count += s.eventCount;
         }
-        return count == 0 ? 0 : total / count;
+        return count == 0 ? Double.NaN : total;
```

Comment:
```suggestion
        return count == 0 ? Double.NaN : total / count;
```

I think you accidentally lost the average computation in the course of the changes.

Reply:
Nice catch!